lib/board.rb

```diff
@@ -0,0 +1,81 @@
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+    reset!
+  end
+  def cells
+    @cells
+  end
+
+  def reset!
+
+     @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+
+  end
+
+
+
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(pos)
+    current_pos = pos.to_i
+
+    cells[current_pos-1]
+
+  end
+
+  def full?
+    @cells.each do |is_full|
+      if is_full.include?(" ")
+        return false
+      end
+    end
+  end
+
+  def turn_count
+    # count = 0
```

Comment:
why don't we just count full space instead that would simplify this code and implicitly return the number for us 

```ruby 
@cells.count { |cell| cell == "X" || cell == "O" }
```